pysiaf/tests/test_nirspec.py

```diff
@@ -114,12 +114,10 @@ def test_against_test_data(siaf=None):
                             elif key_seed == 'rms':
                                 difference_metrics[key_name].append(np.std(test_data['difference_{}'.format(parameter_name)]))
 
-
-
                         print('{} {} SCA_to_OTE transform comparison to {:>10}  tilt={} {:>10} MEAN={:+1.3e} RMS={:1.3e}'.format(sca_name, filter_name, AperName, include_tilt, parameter_name, difference_metrics['diff_{}_{}'.format(parameter_name, 'mean')][index], difference_metrics['diff_{}_{}'.format(parameter_name, 'rms')][index]))
 
-                        assert difference_metrics['diff_{}_{}'.format(parameter_name, 'mean')][index] < 1e-9
-                        assert difference_metrics['diff_{}_{}'.format(parameter_name, 'rms')][index] < 5e-9
+                        assert difference_metrics['diff_{}_{}'.format(parameter_name, 'mean')][index] < 5e-4, "Failed for {}".format(AperName)
```

Comment:
This is quite a significant change in the acceptable tolerance. Does this indicate an issue with the data, the pysiaf code, or something else?

Reply:
I agree that this was weird. I just re-ran this test and got a different result from last time. Now, having both with a threshold of 1e-8 will pass, which makes much more sense. I wonder if something got messed up the last time I ran it. Would you be able to run this test and see if your output agrees with mine below? If it does, I would feel better in saying that I probably just missed something in the above run and can change the thresholds to 1e-8.

```
SCA491 CLEAR SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=False        XAN MEAN=+1.023e-10 RMS=2.521e-09
SCA491 CLEAR SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=False        YAN MEAN=+7.746e-12 RMS=4.146e-10
SCA491 F110W SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=False        XAN MEAN=-3.018e-10 RMS=2.558e-09
SCA491 F110W SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=False        YAN MEAN=-2.197e-11 RMS=4.381e-10
SCA491 F140X SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=False        XAN MEAN=+1.189e-11 RMS=2.718e-09
SCA491 F140X SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=False        YAN MEAN=+3.650e-11 RMS=4.319e-10
SCA492 CLEAR SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=False        XAN MEAN=+1.495e-10 RMS=4.143e-09
SCA492 CLEAR SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=False        YAN MEAN=-5.135e-11 RMS=7.857e-10
SCA492 F110W SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=False        XAN MEAN=-3.256e-10 RMS=4.310e-09
SCA492 F110W SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=False        YAN MEAN=+2.544e-11 RMS=7.447e-10
SCA492 F140X SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=False        XAN MEAN=-2.014e-10 RMS=3.865e-09
SCA492 F140X SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=False        YAN MEAN=-5.413e-11 RMS=6.925e-10
SCA491 CLEAR SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=True        XAN MEAN=-5.224e-10 RMS=2.606e-09
SCA491 CLEAR SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=True        YAN MEAN=+3.331e-10 RMS=4.323e-10
SCA491 F110W SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=True        XAN MEAN=-3.922e-10 RMS=2.804e-09
SCA491 F110W SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=True        YAN MEAN=+3.750e-10 RMS=4.798e-10
SCA491 F140X SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=True        XAN MEAN=-5.949e-10 RMS=2.735e-09
SCA491 F140X SCA_to_OTE transform comparison to NRS1_FULL_OSS  tilt=True        YAN MEAN=+3.598e-10 RMS=3.807e-10
SCA492 CLEAR SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=True        XAN MEAN=-1.453e-10 RMS=4.178e-09
SCA492 CLEAR SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=True        YAN MEAN=+5.095e-10 RMS=6.570e-10
SCA492 F110W SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=True        XAN MEAN=-3.451e-10 RMS=3.892e-09
SCA492 F110W SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=True        YAN MEAN=+5.184e-10 RMS=7.190e-10
SCA492 F140X SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=True        XAN MEAN=-6.541e-10 RMS=4.108e-09
SCA492 F140X SCA_to_OTE transform comparison to NRS2_FULL_OSS  tilt=True        YAN MEAN=+4.694e-10 RMS=6.512e-10
```

Reply:
the generate_nirspec script executes `test_nirspec.test_against_test_data` with a consistent dataset (source_data + SIAF) and it passes for me with 
```
                        assert difference_metrics['diff_{}_{}'.format(parameter_name, 'mean')][index] < 1e-9, "Failed for {}".format(AperName)
                        assert difference_metrics['diff_{}_{}'.format(parameter_name, 'rms')][index] < 5e-9, "Failed for {}".format(AperName)
```

Maybe the issue is that the test is not well designed in the sense that it will fail if we change the source data but not the SIAF that the test is evaluated against? I believe this is the case here where the source data changes but the PRD SIAF is used for the transformations.

Reply:
a possible solution for this test would be to look first for a SIAF.xml in the pre_delivery_data and if none is present default to the PRD SIAF. that way the Travis run should succeed and we are actually testing the right thing.

Alternatively, we could execute the generate script as part of a test.

@shanosborne what do you think?

Reply:
Ah yes, I remember now, the low thresholds I got previously came from running test_nirspec.py, rather than running the test through generate_nirspec.py. That makes sense now.

I agree, having generate_nirspec first look for SIAF.xml in the pre-delivery-data/ directory makes sense to me, since that should always match the source-data/ (and if it doesn't, something is wrong and the test should fail), and only if that data isn't there, would it test against the PRD data. I can re-write test_nirspec to do that and also update the thresholds.

Reply:
Excellent, thanks.